spec/services/price_policies/time_based_price_calculator_spec.rb

```diff
@@ -3,196 +3,323 @@
 require "rails_helper"
 
 RSpec.describe PricePolicies::TimeBasedPriceCalculator do
-
```

Comment:
Relevant changes start in line 200. The rest are only moving the previous tests to a context